runtime/vm/logsupport.c

```diff
@@ -264,42 +265,39 @@ parseLogOptions(char *options, UDATA *optionsFlags, UDATA *parseSucceeded)
 jint
 processXLogOptions(J9JavaVM * vm)
 {
-	IDATA xlogindex;
-	UDATA xlogflags, printhelp, tempflags;
+	IDATA xlogindex = FIND_ARG_IN_VMARGS_FORWARD(OPTIONAL_LIST_MATCH, VMOPT_XSYSLOG_OPT, NULL);
+	/* The default for -Xsyslog is to be turned on and capture 'error' and 'vital' type messages */
+	UDATA xlogflags = defaultLogOptions;
+	UDATA printhelp = FALSE;
+	UDATA tempflags = 0;
 	jint rc = JNI_OK;
 	J9VMInitArgs* j9vm_args = vm->vmArgsArray;
 
 	PORT_ACCESS_FROM_JAVAVM(vm);
 
-	printhelp = FALSE;
-
-	/* The default for -Xlog is to be turned on and capture 'error' and 'vital' type messages */
-	xlogflags = defaultLogOptions;
-
-	xlogindex = FIND_ARG_IN_VMARGS_FORWARD(OPTIONAL_LIST_MATCH, XLOG_OPT, NULL);
-
 	/* If xlogindex == -1 nothing was specified, so use the default setting from above */
 
 	while (xlogindex >= 0) {
-		IDATA optionsfound;
+		IDATA optionsfound = 0;
 		UDATA optionsparsed = FALSE;
-		char *xlogoptions = NULL;
+		char xlogoptionsbuf[SMALL_STRING_BUF_SIZE];
```

Comment:
Why is this buffer needed (it wasn't before)?

Reply:
In order to be compatible with mapped options, the code was changed to use COPY_OPTION_VALUE(), which requires a buffer, instead of GET_OPTION_VALUE().